podcast/2.18.md

```diff
@@ -0,0 +1,545 @@
+
+And I thought that was one of the things that Kirsten was really hammering on is in a lot of cases ... And we're in a similar boat. We're a PHP, but we're built on an internal framework that is very, very rails-like. So we have many of the problems that a rails app has. 
+
+We have models that are chatty with one another. And, basically, I assume they know everything about what the tables look like for another model. I mean, everything is just twisted together. So, in most cases, the real challenge in breaking up the monolith in these code bases is it's not tearing things out and sticking them in another service. You have much deeper problems. 
+
+The biggest problem is that your separations of concerns don't exist. You literally have to shovel some chunk of code over to the side and build a clear interface to that portion of code. And the first step in all of these breakups, I have to imagine, is doing just that because it's not like we can move a portion of the code to a completely separate service if the boundary is not even defined.
+
+And so anyway, for that reason, I was really excited to hear her talking about their struggles in defining the boundaries more than spinning up the services because that has totally been our experience there. We're way more focused on, "Hey, when we get into this portion of the code, let's worry less about where it's going to live and more about what the boundary is going to be and how we're going to enforce that boundary until it does get moved elsewhere. Because we really have to start to build these separations of concerns within the code base, even within the monolithic code base, if we have any chance of ever sticking some of this core functionality into a separate service."
+
+**Beyang Liu:**:
+
+Totally. So we are coming up on the end of time here, but I was wondering as, the last parting thought, do you have any thoughts or maybe prognostications that you'd like to leave listeners with about the future of developer tools, or web development, or maybe just the path forward at Mailchimp?
+
+**Kelly Norton:**
+
+Yeah, I mean, so developer tools are really interesting because I would just say that … I don't know if this is a forward-looking statement or not, but I when we were doing Google Web Toolkit, and when we did things like Speed Tracer, and we really focused on developer tools, and then we left Google intent on starting a business, we intentionally avoided developer tools because we felt that having worked in that space, that it was a thing that nobody paid for.
```

Comment:
```suggestion
Yeah, I mean, so developer tools are really interesting because I would just say that … I don't know if this is a forward-looking statement or not, but when we were doing Google Web Toolkit, and when we did things like Speed Tracer, and we really focused on developer tools, and then we left Google intent on starting a business, we intentionally avoided developer tools because we felt that having worked in that space, that it was a thing that nobody paid for.
```